test/test_xla_backend.py

```diff
@@ -0,0 +1,442 @@
+import os
+import pytest
+import random
+import re
+import string
+import torch
+import torch.distributed as dist
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.core.xla_env_vars as xenv
+import torch_xla.distributed.xla_backend
+
+from contextlib import contextmanager
+from datetime import timedelta
+from pathlib import Path
+
+os.environ[xenv.WORLD_SIZE] = '1'
+os.environ[xenv.ORDINAL] = '0'
+os.environ[
+    xenv.
+    DEVICE_MAP] = 'CPU:0;/job:localservice/replica:0/task:0/device:XLA_CPU:0'
+os.environ['XLA_SAVE_TENSORS_FMT'] = 'hlo'
+os.environ['XLA_BACKEND_BLOCKING_CC_OPS'] = 'True'
+
+
+def get_process_group_xla(rank, size):
+  pg_xla_creator = dist.Backend._plugins[dist.Backend.XLA]
+  pg_xla = pg_xla_creator(
+      prefix_store=None, rank=rank, size=size, timeout=timedelta(minutes=1))
+  return pg_xla
+
+
+def random_filename():
+  return ''.join(random.choices(string.ascii_letters + string.digits, k=12))
+
+
+@contextmanager
+def hlo_dump_monitored(hlo_pattern, match_times=1):
+  if 'XLA_SAVE_TENSORS_FILE' not in os.environ:
+    os.environ['XLA_SAVE_TENSORS_FILE'] = random_filename() + '.hlo'
+  try:
+    yield
+  finally:
+    path = Path(os.environ['XLA_SAVE_TENSORS_FILE'])
+    assert path.exists()
+    hlo = path.read_text()
+    # Note here these measures doesn't guarentee the temp file content
+    # being deleted -- the C++ layer will keep flushing in the content
+    # from last run. The only workaround I found is using pytest multi
+    # workers so each worker only run one test.
+    os.remove(os.environ['XLA_SAVE_TENSORS_FILE'])
+    os.sync()
+    assert not path.exists()
+    matches = re.findall(hlo_pattern, hlo)
+    assert len(list(matches)) == match_times, hlo
+
+
+def hlo_matches(hlo, expected_pattern, match_times=1):
+  matches = re.findall(expected_pattern, hlo)
+  assert len(list(matches)) == match_times, hlo
+
+
+def init_torch_distributed():
+  succeeded = False
+  retries = 5  # retry for port collision.
+  dist.default_pg_timeout = timedelta(minutes=1)
+  for _ in range(retries):
+    try:
+      filename = random_filename()
+      port = random.randrange(49152, 65535)
+      os.environ['XRT_WORKERS'] = f'localservice:0;grpc://localhost:{port}'
+      # `world_size` needs to be 1, so that the rendezvous would only
+      # wait for 1 participant, which is this program itself.
+      # `init_method` is needed to provide a internal storage to comfort
+      # torch.distributed.
+      dist.init_process_group(
+          'xla', rank=0, world_size=1, init_method='file:///tmp/' + filename)
+      succeeded = True
+      break
+    except RuntimeError as err:
+      os.remove('/tmp/' + filename)
```

Comment:
How are we going to clear this temp file if init is successful?

Reply:
Yes this will leave garbage in `/tmp/`. We can try to clean them up if you believe it's necessary.